### DISCUSSION TITLE
How to use auto-imported composable in `nuxt.config.ts`

### DISCUSSION THREAD

**Discussion by ExEr7um** [nuxt/nuxt#20015] [Questions] [answered]
I have a nuxt module that have some auto-imported composables. They work fine in my app, but I want to use one of them in `nuxt.config.ts`.

I think I need to import it manually, but I don't know where from. VS Code suggests `#imports` and `./.nuxt/imports` but then I have an error.

```sh
Cannot restart nuxt:  Cannot find module '#imports'                                                                                                                                       
```

So how can I use auto-imported composables in config?

**Comment by TheAlexLichter** [comment, marked as answer]
Why would you use it there?

No, you cannot use composable "like this" in your Nuxt config. But you could write a util function you'd import and use (manually).

**Reply by ExEr7um** [reply]
I use https://github.com/funkenstudio/sitemap-module-nuxt-3 to generate sitemap. I provide it with function to get dynamic routes like mentioned in the docs and everything works fine.

```ts
import dynamicRoutes from './helpers/dynamicRoutes'
...
sitemap: {
    routes: dynamicRoutes,
},
...
```

But I created a Nuxt module for my organization with shared code, like some composables and components. I wanted to include this function that generates dynamic routes in the module to then import it in all of our projects. But I can't import it properly from the module.

### My solution

I converted this function to serverHandler, so my module generates the endpoint with dynamic routes. Then in my app I created a function that fetches the endpoint and I can import that function in config and use it.

```ts
export default async () => {
  return await $fetch("/api/sitemap-routes")
}
```
